example/heatEquation2D/src/heatEquation2D.cpp

```diff
@@ -0,0 +1,194 @@
+
+    constexpr alpaka::Vec<Dim, Idx> numChunks{
+        (extent[0] - haloSize[0] - 1) / chunkSize[0] + 1,
+        (extent[1] - haloSize[1] - 1) / chunkSize[1] + 1};
+
+    static_assert(
+        (extent[0] - haloSize[0]) % chunkSize[0] == 0 && (extent[1] - haloSize[1]) % chunkSize[1] == 0,
+        "Domain must be divisible by chunk size");
+
+    auto const pitchCurrAcc{alpaka::getPitchesInBytes(uCurrBufAcc)};
+
+    alpaka::WorkDivMembers<Dim, Idx> workDiv_manual{numChunks, threadsPerBlock, elemPerThread};
+
+    StencilKernel<chunkSizeWithHalo.prod()> stencilKernel;
+    BoundaryKernel boundaryKernel;
+
+    for(uint32_t step = 1; step <= numTimeSteps; ++step)
+    {
+        // Compute next values
+        alpaka::exec<Acc>(
+            queue1,
+            workDiv_manual,
+            stencilKernel,
+            uCurrBufAcc.data(),
+            uNextBufAcc.data(),
+            chunkSize,
+            pitchCurrAcc,
+            dx,
+            dy,
+            dt);
+
+        // apply boundaries
+        alpaka::exec<Acc>(
+            queue1,
+            workDiv_manual,
+            boundaryKernel,
+            uNextBufAcc.data(),
+            chunkSize,
+            pitchCurrAcc,
+            step,
+            dx,
+            dy,
+            dt);
+
+        if(step % 100 == 0) // even steps will have currBufHost and PCurr pointing to same buffer
+        {
+            alpaka::memcpy(queue2, uBufHost, uCurrBufAcc);
+            alpaka::wait(queue2);
+            writeImage(step - 1, uBufHost, extent);
+        }
+
+        // So we just swap next to curr (shallow copy)
+        alpaka::wait(queue1);
+        std::swap(uNextBufAcc, uCurrBufAcc);
+    }
+
+    // Copy device -> host
+    alpaka::memcpy(queue1, uBufHost, uCurrBufAcc);
+    alpaka::wait(queue1);
+
+    // Validate
+    auto const [resultCorrect, maxError] = validateSolution(uBufHost, extent, dx, dy, tMax);
+
+    if(resultCorrect)
+    {
+        std::cout << "Execution results correct!" << std::endl;
+        return EXIT_SUCCESS;
+    }
+    else
+    {
+        std::cout << "Execution results incorrect: Max error = " << maxError << " (the grid resolution may be too low)"
+                  << std::endl;
+        return EXIT_FAILURE;
+    }
+}
+
+auto main() -> int
+{
+    // Execute the example once for each enabled accelerator.
+    // If you would like to execute it for a single accelerator only you can use the following code.
+    //  \code{.cpp}
+    //  auto tag = TagCpuSerial;
+    //  return example(tag);
+    //  \endcode
+    //
+    // valid tags:
+    //   TagCpuSerial, TagGpuHipRt, TagGpuCudaRt, TagCpuOmp2Blocks, TagCpuTbbBlocks,
+    //   TagCpuOmp2Threads, TagCpuSycl, TagCpuTbbBlocks, TagCpuThreads,
+    //   TagFpgaSyclIntel, TagGenericSycl, TagGpuSyclIntel
+    return alpaka::executeForEachAccTag([=](auto const& tag) { return example(tag); });
```

Comment:
```suggestion
    return alpaka::executeForEachAccTag([](auto const& tag) { return example(tag); });
```
We're not capturing anything, are we?

Reply:
No we aren't. This is a change which can be made across all alpaka examples.